mktrayicon.c

```diff
@@ -21,38 +21,32 @@ char* strncpy_esc(char *dest, const char *src, size_t n)
 {
 	size_t i = 0;
 	size_t index = 0;
+	size_t outQuoted = 1;
 	while (i < n && src[i] != '\0'){
-		if (src[i] == '\''){ 
-			/* 
-			 * this a real single quote; can't have been escaped
-			 * so we move accordingly, copying everything
-			 */
+		if (src[i] == '\''){
+			outQuoted ^= 1; //xor
 			i++;
-			while (src[i] != '\'' && src[i] != '\0'){
-				// here we escape only single quotes
-				if (src[i] == '\\' && src[i+1] == '\''){
-					dest[index] = '\'';
-					index++;
-					i+=2;
-				}
-				else{
-					dest[index] = src[i];
-					index++;
-					i++;
-				}
+		}
+		else if (src[i] == '\\'){
+			if( (outQuoted && src[i+1] != '\0') || (!outQuoted && src[i+1] == '\'') ){
+				dest[index] = src[i+1];
+				index++;
+				i+=2; 
+			}
+			else if (!outQuoted){ // non escaping '\' in quotes
```

Comment:
This is weird to me. Why should `\` behave differently if you are inside or outside quotes?

Reply:
If you are inside quotes you don't escape any characters (except the quote characters if needed). So a `\` is a character you want to print. If you are outside you escape. That was the whole point with giving the option to quote arguments. Or have I misunderstood something?

Reply:
Ah, I was thinking of `'` as a way to use characters like `|` and `,` inside of a given label or action without having to escape them. I don't know whether or not it should also "turn off" `\` escaping for non-`'` characters.

Thinking some more about this, another option is to forego quoting altogether here, and _just_ provide `\` as a way to escape `\` and `,`. What do you think?

Reply:
Well, I guess we could leave it as is. Another option doesn't hurt anyone. And the escaping with just `\` works from the beginning, so it is ok.

Reply:
It would make the code diff a _lot_ smaller and nicer to read. Since _just_ `\` escaping is much less code, and sufficient to solve the problem, I'm starting to lean towards just doing that and removing the quoting code. How do you feel about that?

Reply:
I thought it was enough to begin with. So, we can revert back to the commit implementing just that. It will be a bit of a pain to check if I made any useful additions while implementing the "quote" code. Let me see what I can do

Reply:
Ok we are back to the commit before I added quote support. I made a copy of the quote code in the `quotes` branch just in case.

Reply:
I will have to make some changes in `master` now. I see some errors